File: 000001-000500/000136.只出现一次的数字/136.py
from typing import List

# 方法一：用时 44ms
# 执行用时超过了 61%
# 消耗内存超过了 71%
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        result = 0
        for i in range(0, len(nums)):
            result ^= nums[i]
        return result


s = Solution()

print(s.singleNumber([2, 2, 1]))
print(s.singleNumber([4, 1, 2, 1, 2]))
print(s.singleNumber([6, 6, 3, 3, 2, 2, 0]))

File: 000001-000500/000136.只出现一次的数字/136.js
/**
 * @param {number[]} nums
 * @return {number}
 */
// 方法一：用时 76ms
// 执行用时超过了 79%
// 消耗内存超过了 75%
var singleNumber = function(nums) {
    let result = 0
    for (let i = 0; i < nums.length; i++) {
        result ^= nums[i]
    }
    return result
};

console.log(singleNumber([2, 2, 1]))
console.log(singleNumber([4, 1, 2, 1, 2]))
console.log(singleNumber([6, 6, 3, 3, 2, 2, 0]))
